documentation/src/js_console.md

```diff
@@ -0,0 +1,110 @@
+
+# Forest JavaScript Console
+
+Forest console stands as an alternative to tools like [Curl](https://github.com/curl/curl) or other subcommands found in `forest-cli` for interacting with the Filecoin JSON-RPC API.
+
+## Starting the console
+
+`forest-cli attach` can be used to open a Javascript console connected to your Forest node.
+
+Like for some other `forest-cli` subcommands you will need to pass an admin `--token` given what endpoints you will call.
+
+For a description of different options please refer to the developer documentation [CLI page](https://github.com/ChainSafe/forest/blob/main/documentation/developer_documentation/CLI.md#cli).
+
+## Interactive Use
+
+First start Forest node inside another terminal:
+
+```bash
+forest --chain calibnet --encrypt-keystore false
+```
+
+To attach to your Forest node, run `forest-cli` with the `attach` subcommand:
+
+```bash
+forest-cli --token <TOKEN> attach 
+```
+
+You should now see a prompt and be able to interact:
+
+```                                          
+Welcome to the Forest Javascript console!
+
+To exit, press ctrl-d or type :quit
+> console.log("Forest running on " + chainGetName())
+Forest running on calibnet
+```
+
+You can directly call JSON-RPC API endpoints that are bound to the console.
+
+For example, `Filecoin.ChainGetName` is bound to the global `chainGetName` function.
+
+### Tips
+
+- Console history is saved in your `~/.forest_history`
+- Use `:clear` to erase session commands before exiting
+- Use `_BOA_VERSION` to get engine version
+
+## Non-interactive Use
+
+### Exec Mode
+
+It is also possible to execute commands non-interactively by passing `--exec` flag and a JavaScript snippet to `forest-cli attach`. The result is displayed directly in the terminal rather than in the interactive console.
+
+For example, to display the current epoch:
+
+```bash
+forest-cli attach --exec "syncStatus().ActiveSyncs[0].Epoch"
+```
+
+Or print node version:
+
+```bash
+forest-cli attach --exec "commonVersion()"
+```
+
+## Builtins
+
+### Helpers
+
+Forest console comes with a number of helper functions that make interacting with Filecoin API easy:
+ - `showPeers()`
+ - `getPeer(peerID)`
+ - `disconnectPeers(count)`
+ - `isPeerConnected(peerID)`
+ - `showWallet()`
+ - `showSyncStatus()`
+ - `sendFIL(to, attoAmount)`
+
+### Timers
+
+In addition, to support part of the JavaScript language, the console also provides implementation for `sleep(seconds)` timer and a tipset based timer, `sleepTipsets(epochs)`, which sleeps till the number of new tipsets added is equal to or greater than `epochs`.
```

Comment:
```
> sleep(5)
Uncaught ReferenceError: sleep is not defined
> sleepTipsets(2)
Uncaught ReferenceError: sleepTipsets is not defined
```

Reply:
Sorry, this is only implemented in the shutdown PR (some kind of follow up). I can move the documentation there.